lib/psulib_traject/workers/hourly_indexer.rb

```diff
@@ -4,15 +4,40 @@ module PsulibTraject
   module Workers
     class HourlyIndexer < Base
       def perform
-        target = Dir.glob("#{ConfigSettings.symphony_data_path}/#{ConfigSettings.symphony_hourlies_subdir}/**/*.m*rc")
+        target = Dir.glob(
+          "#{ConfigSettings.symphony_data_path}/#{ConfigSettings.symphony_hourlies_subdir}/**/*.m*rc"
+        )
         indexed_files = redis.keys('hr:*').map { |e| e.gsub('hr:', '') }
         files_to_index = target - indexed_files
 
+        indexer.logger.info "Found #{files_to_index.length} files to index"
+        array_of_files = files_to_index.map { |file| File.new(file) }
+
+        indexer.process array_of_files
+
         files_to_index.each do |t|
-          puts "indexing #{t}"
-          PsulibTraject::Workers::Indexer.perform_async(t)
+          indexer.logger.info "marking #{t} as done"
           redis.set("hr:#{t}", true)
         end
+
+        deletes = Dir.glob(
+          "#{ConfigSettings.symphony_data_path}/#{ConfigSettings.symphony_hourlies_subdir}/**/*_deletes_*.txt"
+        )
+        processed_deletes = redis.keys('hr:*').map { |e| e.gsub('hr:', '') }
+        files_to_process = deletes - processed_deletes
+
```

Comment:
This method is getting long. I would put all the deleting in a separate method, or even a different worker so we can call the deleting independently of the indexing, if we need to.